### PR TITLE
add stabilityai/FreeWilly2

### PR DESCRIPTION
**Pull request by aniketmaurya**

### CODE REVIEW

File: lit_gpt/config.py [hunk 1]
```diff
@@ -498,4 +498,31 @@ def norm_class(self) -> Type:
         configs.append(copy)
 
 
+###############
+# Stability AI FreeWilly2
+###############
```
Comment (carmocca) [inline lit_gpt/config.py:503]:
```suggestion
##########################
# Stability AI FreeWilly2
##########################
```

**Review by carmocca — COMMENTED** [review]
Since this is a chat model, can you add support for the expected prompt format in `chat/base.py`?

**Comment by aniketmaurya** [comment]
> Since this is a chat model, can you add support for the expected prompt format in `chat/base.py`? For example, this is Llama-2's: https://github.com/Lightning-AI/lit-gpt/blob/main/chat/base.py#L265-L277
> 
> You can find it in the [model card](https://huggingface.co/stabilityai/FreeWilly2)

done! also added tutorial

File: tutorials/download_freewilly_2.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+python scripts/download.py | grep FreeWilly2
+```
+
+which will print
+
+```text
+stabilityai/FreeWilly2
+```
+
+In order to use a specific checkpoint, for instance [FreeWilly2](https://huggingface.co/stabilityai/FreeWilly2), download the weights and convert the checkpoint to the lit-gpt format.
```
Comment (aniketmaurya) [inline tutorials/download_freewilly_2.md:19]: they also have Freewilly1 (Llama 1 I guess). Should we remove this?

Comment (carmocca) [reply tutorials/download_freewilly_2.md:19]: There are no options to choose, there's just one so i agree with removong it

File: chat/base.py [hunk 3]
```diff
@@ -276,6 +276,16 @@ def prompt_config(checkpoint_dir: Path, tokenizer: Tokenizer) -> Tuple[str, Tupl
         stop_tokens = ([tokenizer.eos_id],)
         return system_prompt, stop_tokens
 
+    if re.search("FreeWilly2", checkpoint_name):
+        system_prompt = (
+            "### System:\nThis is a system prompt, please behave and help the user.\n\n"
+            "### User:"
```
Comment (carmocca) [inline chat/base.py:282]: Missed this?
```suggestion
            "### User:\n"
```

File: chat/base.py [hunk 4]
```diff
@@ -276,6 +276,16 @@ def prompt_config(checkpoint_dir: Path, tokenizer: Tokenizer) -> Tuple[str, Tupl
...
+            "{{prompt}}\n\n"
```
Comment (carmocca) [inline chat/base.py:283]:
```suggestion
            "{prompt}\n\n"
```

**Review by carmocca — APPROVED** [review]

**Comment by aniketmaurya** [comment]
updated the PR based on the suggestions! cc: @lantiga @carmocca

**Review by lantiga — APPROVED** [review]

**Comment by m0saan** [comment]
Hi there, @aniketmaurya ! I'm wondering if there's a method to understand the connections between the fields in the `config.json` of Hugging Face and the fields in the `lit_gpt/config.py` Config dataclass. Thanks a lot in advance! :p

**Pull request merged**